### ISSUE TITLE
Event Listing: does not show a valid 6/27 upcoming recurrence (when nearer-future recurrence is canceled)

### ISSUE THREAD

**Issue by jilladams** [department-of-veterans-affairs/va.gov-cms#9455]
## Describe the defect
A recurring event on production is set to recur weekly on Mondays. 
The next calendar upcoming event would be June 20, but that recurrence was canceled. 
The next recurrence is June 27. 
In Upcoming events, the June 27 recurrence (as next upcoming) should show, but does not.

## Screenshots
**Node recurrence settings:** https://prod.cms.va.gov/node/42754/edit
<details>

![Screen Shot 2022-06-14 at 2 03 27 PM](https://user-images.githubusercontent.com/85581471/173688761-7bf0c8a8-da3a-459e-bfeb-b2cb4d0cbe6b.png)
</details>

**Node recurrences in CMS:** 
<details>

![Screen Shot 2022-06-14 at 2 03 35 PM](https://user-images.githubusercontent.com/85581471/173688784-6229a942-e36d-455e-acce-21c93d3508fa.png)
</details>

**Event listing page - missing June 27 recurrence:**  https://www.va.gov/minneapolis-health-care/events/

<details>

![Screen Shot 2022-06-14 at 2 04 09 PM](https://user-images.githubusercontent.com/85581471/173688802-c4d10637-68af-45ae-8c26-839076c4304b.png)
</details>

## To Reproduce
To repro: 
* Visit https://prod.cms.va.gov/node/42754, and see recurrences show next active recurrence on June 27
* Visit https://prod.cms.va.gov/node/42754/edit, and see 6/20 recurrence was canceled, so June 27 is next active
* Vist https://www.va.gov/minneapolis-health-care/events/ and in date order, look for June 27 "MOVE! Start SMART introductory visit (virtual)" event, which is not shown

## Expected behavior
The most upcoming valid (not canceled) recurrence of an event should show in the Event listing on both VAMC and Outreach & Events Hub events.

 ## Additional context
I suspect this may be happening due to the cancelled recurrence, which would otherwise be the next recurrence shown in the Event listing. To prove that out you could: 
* create a weekly recurring event
* verify that the first recurrence shows in event listing page
* cancel the soonest recurrence
* Verify behavior in the listing page / how it's affected. 

I'll try to do this as well and update ticket, but testing on staging is not ideal due to content publishing timelines/overwrite by prod data, etc.

**ALSO:**
Previously, the logic around how recurrences display in upcoming events was modified by https://github.com/department-of-veterans-affairs/va.gov-team/issues/37833.  No PR linked to that ticket, but @ethanteague can point to the previous changes / where logic may need to be adjusted.

## Labels
(You can delete this section once it's complete)
- [x] Issue type (red) (defaults to "Defect")
- [ ] CMS subsystem (green)
- [ ] CMS practice area (blue)
- [x] CMS workstream (orange) (not needed for bug tickets)
- [ ] CMS-supported product (black)

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [ ] `⭐️ Sitewide CMS`
- [X] `⭐️ Public Websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by swirtSJW** [comment]
This may be resolved by #9456   OR it might be a problem with graphQL not surfacing cancelled.  But it would not be easy to check until 9456 is resolved.

**Comment by jilladams** [comment]
@maxx1128 Were you able to determine if this issue repros after your changes in #9456 ?

**Comment by maxx1128** [comment]
@jilladams I haven't been able to determine it, I've been having some issues accessing the PR's environment. I'm leaving the link to it here if it works for you or anyone else with eyes on this ticket - http://eb2dcd8388bacb4ab4f37599abc27e0c.review.vetsgov-internal/

**Comment by jilladams** [comment]
It's a little tricky now that the 6/27 occurrence is in the past, and the series it belongs to has been archived: https://prod.cms.va.gov/minneapolis-health-care/events/42754

Is there a way to modify the CMS data on the test environment? If so, could unarchive that event to verify, and check via this view: 

https://www.va.gov/minneapolis-health-care/events/?endDateDay=29&endDateMonth=06&endDateYear=2022&selectedOption=custom-date-range&startDateDay=12&startDateMonth=06&startDateYear=2022

**Comment by jilladams** [comment]
The alternative is to create an event with recurrence that starts right now, cancel one of the recurrences, and check if the rest show up appropriately. But again: not sure how we deal with CMS data for test envs.

**Comment by jilladams** [comment]
Example event we can use to repro: https://prod.cms.va.gov/minneapolis-health-care/events/42754

Per Swirt: this could be unarchived in a tuboat environment and then trigger a content release.

That can get sorted when this ticket is picked up, by whomever. If you're not sure how to do it, flag as a blocker in scrum and we'll get it unblocked.

**Comment by jtmst** [comment]
Verified by observing CMS data changed in tugboat environments using the following steps:

1. Verify that events page listing is properly displaying the next event.
2. Go to CMS Edit page and click "edit event series"
3. Go to the next event listed and edit it so that it is cancelled. 
4. Publish to tugboat CMS instance
5. Point your local content-build to the tugboat CMS instance and rebuild
6. Go to events page and verify that the cancelled event is not displayed, but that the following event is

<img width="1005" alt="Screen Shot 2022-07-11 at 1 27 51 PM" src="https://user-images.githubusercontent.com/61624970/178325239-88f9ece2-6ccd-40dd-9a54-8ae07961d3eb.png">

**Comment by jilladams** [comment]
Nice! Thanks for that, Josh. Closing.
